Mathlib/Analysis/Analytic/Within.lean

```diff
@@ -0,0 +1,367 @@
+
+/-!
+# Properties of analyticity restricted to a set
+
+From `Mathlib.Analysis.Analytic.Basic`, we have the definitons
+
+1. `AnalyticWithinAt 𝕜 f s x` means a power series at `x` converges to `f` on `𝓝[s] x`, and
+    `f` is continuous within `s` at `x`.
+2. `AnalyticWithinOn 𝕜 f s t` means `∀ x ∈ t, AnalyticWithinAt 𝕜 f s x`.
+
+This means there exists an extension of `f` which is analytic and agrees with `f` on `s ∪ {x}`, but
+`f` is allowed to be arbitrary elsewhere.  Requiring `ContinuousWithinAt` is essential if `x ∉ s`:
+it is required for composition and smoothness to follow without extra hypotheses (we could
+alternately require convergence at `x` even if `x ∉ s`).
+
+Here we prove basic properties of these definitions. Where convenient we assume completeness of the
+ambient space, which allows us to related `AnalyticWithinAt` to analyticity of a local extension.
+-/
+
+noncomputable section
+
+open Topology Filter ENNReal
+
+open Set Filter
+
+variable {α : Type*}
+
+variable {𝕜 : Type*} [NontriviallyNormedField 𝕜]
+
+variable {E F G H : Type*} [NormedAddCommGroup E] [NormedSpace 𝕜 E] [NormedAddCommGroup F]
+  [NormedSpace 𝕜 F] [NormedAddCommGroup G] [NormedSpace 𝕜 G] [NormedAddCommGroup H]
+  [NormedSpace 𝕜 H]
+
+/-!
+### Basic properties
+-/
+
+@[simp] lemma hasFPowerSeriesWithinOnBall_univ {f : E → F} {p : FormalMultilinearSeries 𝕜 E F}
+    {x : E} {r : ℝ≥0∞} :
+    HasFPowerSeriesWithinOnBall f p univ x r ↔ HasFPowerSeriesOnBall f p x r := by
+  constructor
+  · intro h
+    exact ⟨h.r_le, h.r_pos, fun {y} m ↦ h.hasSum (mem_univ _) m⟩
+  · intro h
+    refine ⟨h.r_le, h.r_pos, fun {y} _ m => h.hasSum m, ?_⟩
+    exact (h.continuousOn.continuousAt (EMetric.ball_mem_nhds x h.r_pos)).continuousWithinAt
+
+@[simp] lemma hasFPowerSeriesWithinAt_univ {f : E → F} {p : FormalMultilinearSeries 𝕜 E F} {x : E} :
+    HasFPowerSeriesWithinAt f p univ x ↔ HasFPowerSeriesAt f p x := by
+  simp only [HasFPowerSeriesWithinAt, hasFPowerSeriesWithinOnBall_univ, HasFPowerSeriesAt]
+
+@[simp] lemma analyticWithinAt_univ {f : E → F} {x : E} :
+    AnalyticWithinAt 𝕜 f univ x ↔ AnalyticAt 𝕜 f x := by
+  simp only [AnalyticWithinAt, hasFPowerSeriesWithinAt_univ, AnalyticAt]
+
+lemma analyticWithinOn_univ {f : E → F} :
+    AnalyticWithinOn 𝕜 f univ ↔ AnalyticOn 𝕜 f univ := by
+  simp only [AnalyticWithinOn, analyticWithinAt_univ, AnalyticOn]
+
+lemma HasFPowerSeriesWithinAt.continuousWithinAt {f : E → F} {p : FormalMultilinearSeries 𝕜 E F}
+    {s : Set E} {x : E} (h : HasFPowerSeriesWithinAt f p s x) : ContinuousWithinAt f s x := by
+  rcases h with ⟨r, h⟩
+  exact h.continuousWithinAt
+
+lemma AnalyticWithinAt.continuousWithinAt {f : E → F} {s : Set E} {x : E}
+    (h : AnalyticWithinAt 𝕜 f s x) : ContinuousWithinAt f s x := by
+  rcases h with ⟨p, h⟩
+  exact h.continuousWithinAt
+
+/-- `AnalyticWithinAt` is trivial if `𝓝[s] x = ⊥` -/
+lemma analyticWithinAt_of_eq_bot {f : E → F} {s : Set E} {x : E} (h : 𝓝[s] x = ⊥) :
```

Comment:
It should be true under the weaker condition that neighborhoods of `x` within `s` are contained inside `{x}`.

Reply:
Fixed, thanks!